plugin/file/lookup.go

```diff
@@ -239,8 +255,8 @@ func (z *Zone) Lookup(ctx context.Context, state request.Request, qname string)
 
 	// Hacky way to get around empty-non-terminals. If a longer name does exist, but this qname, does not, it
 	// must be an empty-non-terminal. If so, we do the proper NXDOMAIN handling, but set the rcode to be success.
-	if x, found := tr.Next(qname); found {
-		if dns.IsSubDomain(qname, x.Name()) {
+	if nextElem != nil {
```

Comment:
That this logic has essentially changed from `tr.Next(qname)` to `tr.Next(parts)`, seems like it might break empty non-terminal resolution.

Reply:
> That this logic has essentially changed from `tr.Next(qname)` to `tr.Next(parts)`, seems like it might break empty non-terminal resolution.

I don't see an existing test that validates queries for empty non-terminals should get a 'NOERROR' response. Should add it if one doesn't currently exist. e.g. in `lookup_test.go`...

`dbMiekNL` +=
```
test.ent	IN	A	139.162.196.79
```

and `dnsTestCases` +=
```
	{
		Qname: "ent.miek.nl.", Qtype: dns.TypeA,
		Ns: []dns.RR{
			test.SOA("miek.nl.	1800	IN	SOA	linode.atoom.net. miek.miek.nl. 1282630057 14400 3600 604800 14400"),
		},
	},
```

Reply:
Hi @chrisohaver I reverted back the tr.Next(qname) and added a new test case in lookup_test.go.

Reply:
Cool - thanks.  Looks like the other two wildcard related tests in azure and clouddns are passing now after that change too.

Reply:
Hi @chrisohaver, the idea is to iteratively find the next record that is the subdomain of the parts, it will finally find the last matching record.

For the zone file in `cornerCasesWildcard`
<pre>
     org
      |
   example
      |
      d
     / \
    c   *
    |
    r
</pre>

`example.org.`, ne is `*.d.example.org.`
`d.example.org.` ne is `*.d.example.org.`

`c.d.example.org.` ne is `r.c.d.example.org.`
`r.c.d.example.org.` ne is `r.c.d.example.org.`
...

`something.d.example.org.` ne is `*.d.example.org.` (inherent from `d.example.org.` as `*.d.example.org.` is not a child of `something.d.example.org.`)
...

If number of labels of ne is greater than number of labels of the matching wildcard, we want to return NXDOMAINS